### ISSUE TITLE
Field type loses array kind if get/set field options are defined

### ISSUE THREAD

**Issue by mfulop** [typegoose/typegoose#422]
## Versions

- NodeJS: 12.18.0
- Typegoose(NPM): 7.4.2 
- mongoose: 5.10.18

## What is the Problem?

Field type loses array kind if get/set field options are defined.

## Code Example

```ts
  @prop({ type: Schema.Types.String, default: [] })
  public good?: string[] // will be string[]

  @prop({ type: Schema.Types.String, default: [], get: _.identity, set: _.identity })
  public bad: string[] // will be string
```

## Do you know *why* it happens?

utils.initProperty returns array (correctly) but if options get/set are defined then the type is reset here:
https://github.com/typegoose/typegoose/blob/39e5d0c2050b500bc3255438cf8fefa44c652b13/src/internal/processProp.ts#L182

**Comment by hasezoey** [comment]
true, currently it dosnt differentiate between normal prop and array

https://github.com/typegoose/typegoose/blob/57e147c602319f23a0c092286536ddfd8a5bd3ce/src/internal/processProp.ts#L173-L189

**Comment by hasezoey** [comment]
included in 7.4.4

**Comment by hasezoey** [comment]
for anyone reading later, [the documentation](https://typegoose.github.io/typegoose/docs/api/decorators/prop#get--set) got extended to show to to have differing runtime times from what is stored in the database
